### PR TITLE
Search alt

### PR DESCRIPTION
**Pull request by RadekW86**
Please, review the Search functionality. 
All the significant developments are described in the names of the commits.
Please test all the possible strange user behaviour to find loopholes.

### CODE REVIEW

File: src/TopBar/Search/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    input === "" ? dispatch(disengage()) : dispatch(engage());
+  }, [input]);
+
+  const onInputChange = ({ target }) => {
+    setInput(target.value.trim());
+    setQueryParameters({
+      nameValue: "page",
+      value: page,
+      nameValue2: "search",
+      value2: target.value.trim() !== "" ? target.value : undefined,
```
Comment (AngelikaKrygier) [inline src/TopBar/Search/index.js:35]: What do you think about it? : ) 

value2: target.value.trim() && target.value.trim()

File: src/features/movies/MoviesList/index.js [hunk 2]
```diff
@@ -40,22 +69,47 @@ export const MoviesList = () => {
...
-                name={movie.title || movie.original_name}
-                productionInF={movie.release_date}
-                genres={movie.genre_ids}
-                rate={movie.vote_average}
-                votes={movie.vote_count}
-                id={movie.id}
-              />
-            ))}
-            foot={<Pagination currentPage={page} totalPages="500" />}
+            movies={moviesList.length === 0 ? "" : "movies"}
```
Comment (AngelikaKrygier) [inline src/features/movies/MoviesList/index.js:72]: What do you think about the shorter version? :

movies={moviesList.length ? "movies" : ""}

**Review by AngelikaKrygier — APPROVED** [review]
It was a difficult part of the application, well done!

**Review by JakubJachacz — APPROVED** [review]
Impressive work! I managed to find one little bug, but I'm not sure how to fix it.
The steps to recreate it are as follow:
1. Go to people page and write aaa in search bar (might work with other inputs too)
2. Immediately switch to Movies page.
 Now there should be Popular Movies title with a paper plane image under it instead of tiles. The bug sometimes happens and sometimes doesn't, so it might take a few tries to recreate.

File: src/TopBar/Search/index.js (hunk 1 above)
Comment (RadekW86) [reply src/TopBar/Search/index.js:35]: the purpose of this line is to set exactly undefined to remove "search=" from URL, so I think it should stay the way it is.

File: src/features/movies/MoviesList/index.js (hunk 2 above)
Comment (RadekW86) [reply src/features/movies/MoviesList/index.js:72]: ok

**Pull request merged**